3rdparty/preact/signals-core/index.ts

```diff
@@ -660,4 +660,4 @@ function effect(compute: () => unknown): () => void {
 	return effect._dispose.bind(effect);
 }
 
-export { signal, computed, effect, batch, Signal, ReadonlySignal };
+export { signal, computed, effect, batch, Signal, type ReadonlySignal };
```

Comment:
Most of the identifiers exported on this line are both types _and_ values, but `ReadonlySignal` is just a type. The `type` annotation helps bundlers deal with this correctly ([read more](https://www.typescriptlang.org/tsconfig#exports-of-non-value-identifiers)).

Reply:
Ah I didn't know about this before. A bit more from ChatGPT:

```
Some third-party JavaScript bundlers like webpack and Rollup might not understand TypeScript's semantics around types, since they don't exist in JavaScript. They might see an import statement and assume it's importing a value that needs to be included in the bundle, even if it's actually only importing a type that gets erased during TypeScript's compilation step.

This can lead to unnecessarily larger bundles, or even errors if a module that only exports types is not included in the bundle because the bundler assumes it needs to be there at runtime.

By using import type or export type, you're making it explicit to both TypeScript and any tooling that understands TypeScript that these imports and exports are type-only, and so they don't need to be included in the JavaScript output. This can help improve bundle size and reduce potential errors.

The TypeScript team actually recommends using import type and export type when you're working in a codebase that also gets built with a bundler, just to be on the safe side and to make your intent clearer to other developers.
```